plugins/Live/Live.php

```diff
@@ -114,4 +122,91 @@ public function renderVisitorIcons(&$renderedDetails, $visitorDetails)
             $renderedDetails .= $instance->renderIcons($visitorDetails);
         }
     }
+
+    public static function getSegmentWithVisitorId()
```

Comment:
Maybe this could be moved into a different class outside the Plugin class?

Also is the cache needed here or can we maybe use transient cache so it will be also invalidated between tests etc?

Reply:
Not sure where it would be useful to move that method to.

I guess the cache is there to prevent `Request::processRequest('Live.getMostRecentVisitorId')` being called multiple times with different results. Guess using transient cache should be fine as well...

Reply:
Be very useful to add this as a comment 👍 as we will otherwise eventually introduce a bug here when someone removes the caches (it looks like it isn't needed.)